iroh-api/src/api_ext.rs

```diff
@@ -0,0 +1,105 @@
+use std::path::{Path, PathBuf};
+
+use crate::{Api, Cid, IpfsPath, OutType};
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use futures::Stream;
+use futures::StreamExt;
+use relative_path::RelativePathBuf;
+
+#[async_trait(?Send)]
+pub trait ApiExt: Api {
+    /// High level get, equivalent of CLI `iroh get`
+    async fn get<'a>(
+        &self,
+        ipfs_path: &IpfsPath,
+        output_path: Option<&'a Path>,
+    ) -> Result<PathBuf> {
+        let cid = ipfs_path
+            .cid()
+            .ok_or_else(|| anyhow!("IPFS path does not refer to a CID"))?;
+        let root_path = get_root_path(cid, output_path);
```

Comment:
Ah, okay, we are actually missing a step here that was covered by `make_output_path`

You can see for yourself if you take these steps:
- add a directory "a" with a file inside "b" with content "hello", let's say the cid that results is `bafyfoo`
- run `iroh get bafyfoo/a/b new_b`
- the expected result is to get a new file `./new_b` with content "hello", instead you get `./new_file/a/b`
- you can see the difference if you compare the output from `main` & this branch running the same `get` command

Reply:
this pr shows the correct behaviour: https://github.com/n0-computer/iroh/pull/339

Reply:
Ah, I had missed that `iroh get cid/a/b` is a thing at all! I thought you had to address it all with a cid. Wasn't in the list of test cases in the `iroh get` issue for that reason. I will look at your work and integrate it.

Reply:
I've now modified `get_stream` so it is aware of sub paths and skips anything unrelated. This appears to do the right thing when I test it by hand. 

I'm not very happy that I can't have this behavior under test, but I do think that `get_stream` should deliver only what someone asked for and not more; after that it is just a matter of writing it out. Ideally I think this behavior belongs in the resolver; after all `resolve_recursive_with_paths` knows everything it has to know already and it could potentially be doing optimizations that the API layer can't.